umbral/pre.py

```diff
@@ -108,6 +115,40 @@ def from_bytes(cls, capsule_bytes: bytes, curve: ec.EllipticCurve = None):
         components = splitter(capsule_bytes)
         return cls(*components)
 
+    def _set_cfrag_checking_key(self, key_type, key: UmbralPublicKey):
+        current_key = self._cfrag_checking_keys[key_type]
+
+        if current_key is None:
+            if key is None:
+                raise TypeError("The Delegating Key is not set and you didn't pass one.")
+            else:
+                self._cfrag_checking_keys[key_type] = key
+                return True
+        elif key in (None, self._cfrag_checking_keys[key_type]):
+            return False
+        else:
+            raise ValueError("The Delegating Key is already set; you can't set it again.")
+
+    def set_delegating_key(self, key: UmbralPublicKey):
+        return self._set_cfrag_checking_key("delegating", key)
+
+    def set_encrypting_key(self, key: UmbralPublicKey):
+        return self._set_cfrag_checking_key("encrypting", key)
+
+    def set_verifying_key(self, key: UmbralPublicKey):
+        return self._set_cfrag_checking_key("verifying", key)
+
+    def set_three_keys(self,
+                              delegating_key: UmbralPublicKey = None,
+                              encrypting_key: UmbralPublicKey = None,
+                              verifying_key: UmbralPublicKey = None
+                              ):
+        delegating_key_details = self.set_delegating_key(delegating_key)
+        encrypting_key_details = self.set_encrypting_key(encrypting_key)
+        verifying_key_details = self.set_verifying_key(verifying_key)
+
+        return delegating_key_details, encrypting_key_details, verifying_key_details
```

Comment:
Again, I find it pretty redundant that a `set` method returns what the user just provided?

Reply:
It doesn't anymore - it just returns `True` if it was set new.